### ISSUE TITLE
Derivation PDF

### ISSUE THREAD

**Issue by EngDM** [buddyd16/pyMAOS#1]
Apologies in advance if this is missuse of the issues form.

I am trying to work my way through the linear load derivation, what exactly does the third page represent? Is this an equation for deflection at a point? Is there anything in that derivation would be deflection at point "x" for a linear load?

**Comment by buddyd16** [comment]
Page 3 of that pdf is the simplified symbolic solutions for the integration
constants.

Towards the bottom of page 1 you should see the word Deflection below that
is the piecewise functions for deflection at a point for the first two
bounds x=0->a, x=a->b. At the top of page 2 is the last bound of x=b->L.

**Comment by EngDM** [comment]
So given that there are C# constants in that equation, does this imply that there is no set piecewise function that could be implemented into a spreadsheet or something?

**Comment by buddyd16** [comment]
No, the equations shown in the derivation are the set piecewise functions. The formulas for constants C1-C12 are indicated on page 3.

a bit confused on what you are not understanding the derivation is laid out going from load functions to deflection functions via direct integration. Then initial boundary and compatibility conditions are defined. Then there are a set of 12 equations defined by applying the boundary/compatibility conditions these need to solved simultaneously for the integration constants c1-c12. Page 3 is the resulting solution for the integration constants, these then get plugged into the piecewise functions defined on page 1.

For the implementation in python see the class beginning on line 407 in the loading.py file.

**Comment by EngDM** [comment]
Ah I see, so on page 3 that is what C1-C12 are? I’m assuming broken up by line. Sorry, it’s not labeled and I’m unfamiliar with this derivation. Thanks

**Comment by buddyd16** [comment]
“ Ah I see, so on page 3 that is what C1-C12 are? ”

yes, broken up by line. 

No worries, the derivation itself is fairly straightforward (assuming you’re familiar with basic differential equations and how to apply boundary conditions) the equations just get huge for the linear load case so the hard part is making sure you have typed everything in correctly.

**Comment by EngDM** [comment]
Yea I see that looking at it further. It’s just massive since it’s a catch all for all UDL loads, sloped or constant over any portion of the beam. Anyways, I hope I haven’t been too much of a bother. Thanks,

**Comment by buddyd16** [comment]
No bother at all, sometimes it’s a bit difficult to pin point how exactly to best help with the limited context of these messages. Happy to help if I am able.

**Comment by EngDM** [comment]
Hey again,

I'm in the middle of implementing the point load derivation, however I am noticing that it is only accurate when a>L/2 (or a>b). I've compared my values with 2 different online solvers, and they align when a>b.

I couldn't see anywhere in the derivation that said a has to be larger than b, however in the simplified equations online they are all listed only when x<a and a>b.

**Comment by buddyd16** [comment]
a bit confused there is no b variable in the point load derivation? the point load is fully located by a single position variable, a.

**Comment by EngDM** [comment]
Say b = a-L.

**Comment by EngDM** [comment]
![image](https://github.com/buddyd16/pyMAOS/assets/108320854/4b5a1fd3-1b29-496f-979a-62a179f7923c)

Also, the bottom of the derivation appears to indicate that C6 should be positive?

![image](https://github.com/buddyd16/pyMAOS/assets/108320854/10829980-581b-4fdf-919b-56982ab1faa1)

**Comment by buddyd16** [comment]
so no the initial equation is written with the assumption of a positive C6, the resulting solution shows that the value is actually (-P*a^3)/6.

I've run hand checks and checks against other software and am getting agreement with my functions?

**Comment by EngDM** [comment]
Ah that makes sense.

I'll do some more investigating, might have to re-enter everything. Given that it is only when a<x, it's probably an issue with the constants associated with it.

Just to clarify, it is for any variation of this loading and not just x<L/2 like some other published equations?

**Comment by buddyd16** [comment]
My derivation is for the point load located at any value of a.

**Comment by buddyd16** [comment]
my python class for the point load begins on line 318 here:
https://github.com/buddyd16/pyMAOS/blob/main/pyMAOS/loading.py
